controllers/directional_trading/lp_management.py

```diff
@@ -0,0 +1,562 @@
+import asyncio
+import functools
+import logging
+import time
+from decimal import Decimal
+from enum import StrEnum
+from typing import Any, Awaitable, Callable, Concatenate, ParamSpec, TypeVar
+
+from pydantic.dataclasses import dataclass
+
+from hummingbot.client.settings import GatewayConnectionSetting
+from hummingbot.core.gateway.gateway_http_client import GatewayHttpClient
+from hummingbot.logger import HummingbotLogger
+
+
+@dataclass
+class GatewayStatus:
+    gateway_ready: bool = False
+    wallet_address: str = ""
+
+
+@dataclass
+class GatewayConnectionInfo:
+    connector: str
+    chain: str
+    network: str
+
+
+@dataclass
+class PoolInfo:
+    base_token: str | None
+    quote_token: str | None
+    last_price: Decimal
+
+
+async def check_gateway_status(
```

Comment:
I believe this fn is no longer needed if you are using the GatewayLP class since it inherits from GatewayBase which contains a Gateway status check.